pkg/proxy/iptables/proxier.go

```diff
@@ -555,7 +553,11 @@ func (proxier *Proxier) OnServiceUpdate(allServices []api.Service) {
 		healthcheck.DeleteServiceListener(hc.namespace, hc.nodeport)
 	}
 
-	proxier.syncProxyRules()
+	if !reflect.DeepEqual(newServiceMap, proxier.serviceMap) {
```

Comment:
might be an optimization to compare the len() first and if not equal, skip the reflection, or even to compare keys.